src/App.tsx

```
@@ -1,26 +1,110 @@
+import { useEffect, useState } from 'react';
+import { Loader, TodoFilter, TodoList, TodoModal } from './components';
+
+import { getTodos } from './api';
+
+import { Todo } from './types/Todo';
+
+import { setQuery, setStatus } from './features/filter';
+import { useAppDispatch, useAppSelector } from './hooks/hooks';
+import { setTodos } from './features/todos';
+import { clearCurrentTodo, setCurrentTodo } from './features/currentTodo';
+
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-import { Loader, TodoFilter, TodoList, TodoModal } from './components';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+export const App = () => {
+  const [modal, setModal] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+  const { query, status } = useAppSelector(state => state.filter);
+  const todos = useAppSelector(state => state.todos);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    let isCancelled = false;
+
+    setIsLoading(true);
+
+    async function getTodoData() {
+      try {
+        const todoData = await getTodos();
+
+        if (!isCancelled) {
+          dispatch(setTodos(todoData));
+        }
+      } catch (error) {
```

The catch block is empty. At minimum log or handle the error (e.g. `console.error(error)`) so failures in `getTodos()` are visible during development. Silent failures make debugging harder and may hide issues with loading data. This also helps ensure the UI/state can respond appropriately on error (requirement doesn't mandate UI error display, but logging is important).